File: pm_backup/app/src/main/java/com/arielfaridja/ezrahi/data/typesConverter.java
package com.arielfaridja.ezrahi.data;

import androidx.room.TypeConverter;

import com.arielfaridja.ezrahi.entities.Latlng;
import com.arielfaridja.ezrahi.entities.User;
import com.google.gson.Gson;

import org.json.JSONException;

public class typesConverter {
    @TypeConverter
    public static String userToString(User user) {
        return user == null ? null : user.toString();
    }

    @TypeConverter
    public static User stringToUser(String string) throws JSONException {
        User u;
        Gson gson = new Gson();
        u = gson.fromJson(string, User.class);
        return u;
    }

    @TypeConverter
    public static Latlng stringToLatlng(String string) throws JSONException {
        return Latlng.StringToLatlng(string);
    }

    @TypeConverter
    public static String latlngToString(Latlng latlng) {
        return latlng.toString();
    }
}

File: app/src/main/java/com/arielfaridja/ezrahi/data/TypesConverter.kt
package com.arielfaridja.ezrahi.data

import androidx.room.TypeConverter
import com.arielfaridja.ezrahi.entities.Latlng
import com.arielfaridja.ezrahi.entities.User
import com.google.gson.Gson

class TypesConverter {
    companion object {
        @TypeConverter
        @JvmStatic
        fun userToString(user: User?): String? {
            return user?.toString()
        }

        @TypeConverter
        @JvmStatic
        fun stringToUser(string: String?): User? {
            return if (string == null) null else Gson().fromJson(string, User::class.java)
        }

        @TypeConverter
        @JvmStatic
        fun stringToLatlng(string: String?): Latlng? {
            return if (string == null) null else Latlng.StringToLatlng(string)
        }

        @TypeConverter
        @JvmStatic
        fun latlngToString(latlng: Latlng?): String? {
            return latlng?.toString()
        }

        @TypeConverter
        @JvmStatic
        fun fromIntList(list: ArrayList<Int>?): String? {
            return list?.joinToString(",")
        }

        @TypeConverter
        @JvmStatic
        fun toIntList(data: String?): ArrayList<Int>? {
            return if (data.isNullOrEmpty()) arrayListOf() else ArrayList(data.split(",").map { it.toInt() })
        }
    }
}
